src/test/java/seedu/dengue/testutil/PersonListBuilder.java

```diff
@@ -0,0 +1,152 @@
+        Person defaultPerson = this.personBuilder.build();
+
+        ages.forEach(age -> personList.add(this.personBuilder.withAge(age).build()));
+
+        return new PersonListBuilder(defaultPerson, personList);
+    }
+
+    /**
+     * Returns a new {@code PersonListBuilder} which builds a {@code List} of {@code Person} instances with
+     * postal codes following the list provided, but are otherwise identical to the default {@code Person}.
+     *
+     * @param postals The desired postals of the people in the output list.
+     * @return A new list builder.
+     */
+    public PersonListBuilder withPostals(List<String> postals) {
+        List<Person> personList = new ArrayList<>();
+        Person defaultPerson = this.personBuilder.build();
+
+        postals.forEach(postal -> personList.add(this.personBuilder.withPostal(postal).build()));
+
+        return new PersonListBuilder(defaultPerson, personList);
+    }
+
+    /**
+     * Returns a new {@code PersonListBuilder} which builds a {@code List} of {@code Person} instances with
+     * case dates following the list provided, but are otherwise identical to the default {@code Person}.
+     *
+     * @param dates The desired dates of the cases in the output list.
+     * @return A new list builder.
+     */
+    public PersonListBuilder withDates(List<String> dates) {
+        List<Person> personList = new ArrayList<>();
+        Person defaultPerson = this.personBuilder.build();
+
+        dates.forEach(date -> personList.add(this.personBuilder.withDate(date).build()));
+
+        return new PersonListBuilder(defaultPerson, personList);
+    }
+
+    /**
+     * Returns a new {@code PersonListBuilder} which builds a {@code List} of {@code Person} instances with
+     * dengue variants following the list provided, but are otherwise identical to the default {@code Person}.
+     *
+     * @param variants The desired list of dengue variants (possibly empty) contracted by the people in the output list.
+     * @return A new list builder.
+     */
+    public PersonListBuilder withVariants(List<String[]> variants) {
+        List<Person> personList = new ArrayList<>();
+        Person defaultPerson = this.personBuilder.build();
+
+        variants.forEach(variantArr -> personList.add(this.personBuilder.withVariants(variantArr).build()));
+
+        return new PersonListBuilder(defaultPerson, personList);
+    }
+
+    /**
+     * Returns a {@code List} of {@code Person} instances, according to the list of people stored.
+     *
+     * @return The desired list of people.
+     */
+    public List<Person> build() {
+        return new ArrayList<>(this.personList);
+    }
+
+    /**
+     * Returns a {@code List} containing only one default {@code Person} instance.
+     *
+     * @return A singleton containing the default person.
+     */
+    public List<Person> buildSingle() {
+        return List.of(this.personBuilder.build());
+    }
+
```

Comment:
did you mean one instance of the `List<Person>` or `PersonListBuilder`? Also, in order to render this a singleton class, you'll need to implement other measures to prevent the calling of the constructor again, other than making it private

E.g. maintain a static count/boolean value for whether an object has been initialised. For this reason i don't recommend the use of singletons

Reply:
Thanks for the catch! Sorry, I meant singleton as in a list containing one element. I'll rephrase that now!